aw-server/src/endpoints/key_value.rs

```diff
@@ -0,0 +1,37 @@
+use std::sync::MutexGuard;
+use rocket_contrib::json::Json;
+use rocket::http::Status;
+use rocket::State;
+use crate::endpoints::ServerState;
+
+use aw_datastore::{Datastore, DatastoreError};
+
+#[post("/<key>", data="<message>")]
+pub fn value_new(state: State<ServerState>, key: String, message: Json<String>)
+    -> Result<Status, Status> {
+
+    let data = message.into_inner();
+    let datastore: MutexGuard<'_, Datastore> = endpoints_get_lock!(state.datastore);
+    let result = datastore.create_value(&key, &data);
+    return match result {
+        Ok(_) => Ok(Status::Created),
+        Err(_) => Err(Status::InternalServerError)
```

Comment:
We should also print the error to the log so it's actually debuggable.

Reply:
Changed and fixed this in 6c7dd29